cardinal_toe_app/src/app/components/location-menu/location-menu.component.ts

```diff
@@ -11,44 +11,34 @@ import { NgIf, TitleCasePipe } from '@angular/common';
   templateUrl: './location-menu.component.html',
   styleUrls: ['./location-menu.component.scss']
 })  
+
 export class LocationMenuComponent implements OnChanges {
   @Input() display: boolean = false;
-  @Input() selectedDestination?: Place | undefined;
-  @Input() currentLocation?: boolean | undefined;
+  @Input() selectedRoute!: Location;
+  @Input() isCurrentLocation!: boolean;
   @Output() changeDisplay = new EventEmitter<boolean>();
-  @Output() changeNode = new EventEmitter<Place>();
-
-  localDestination: Place | undefined;
-  localCurrent: boolean | undefined;
-  canSearch!: boolean | undefined;
-  canGo!: boolean | undefined;
+  @Output() newLocation = new EventEmitter<Location>();
 
+  canSearch!: boolean;
+  canGo!: boolean;
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes && this.selectedDestination) {
-      this.localCurrent = this.currentLocation;
-      this.canSearch = !this.localCurrent;
-      this.canGo = this.localCurrent;
-      this.localDestination = this.selectedDestination;
-    } else {
-      this.resetVariables();
+    if (changes && this.selectedRoute) {
```

Comment:
could this be a guard clause? the entire body of the function is wrapped in an `if` that normally means yes.